### PR TITLE
Tech 231

### PR DESCRIPTION
**Pull request by lynderup**
Initial helm chart. Only http routes at the moment

### CODE REVIEW

File: chart/values.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+ports:
+  traefik:
+    port: 8080
+    protocol: TCP
+  web:
+    serviceType: NodePort
+    port: 80
+    nodePort: 32286
+    protocol: TCP
+
```
Comment (jacobcrawford) [inline chart/values.yaml:19]: too many whitespace lines.

File: chart/templates/service.yaml [hunk 2]
```diff
@@ -0,0 +1,45 @@
+# Makes a nested dictionary, from .Values.ports, with the structure ServiceType->PortName->PortConfig
```
Comment (jacobcrawford) [inline chart/templates/service.yaml:1]: Maybe explain the structure of the dictionary in yaml or json format.

File: chart/values.yaml [hunk 3]
```diff
@@ -0,0 +1,22 @@
+image:
+  name: traefik
```
Comment (jacobcrawford) [inline chart/values.yaml:2]: I think we need to specify a version of the traefik image.

File: chart/values.yaml [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+  tag: ""
```
Comment (jacobcrawford) [inline chart/values.yaml:3]: "tag" is not used anywhere.

File: chart/values.yaml [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+  pullPolicy: IfNotPresent
+
+nameOverride: ""
```
Comment (jacobcrawford) [inline chart/values.yaml:6]: The use of nameOverride and fullnameOverride is unclear to me. Can you make a comment infile?

File: chart/templates/service.yaml (hunk 2 above)
Comment (jacobcrawford) [reply chart/templates/service.yaml:1]: Maybe give an example of a yaml tree input and output of this.

**Review by jacobcrawford — CHANGES_REQUESTED** [review]
Looks good. A few questions and change requests.

File: chart/values.yaml (hunk 3 above)
Comment (MartinSchmidt) [reply chart/values.yaml:2]: I agree.

File: chart/values.yaml (hunk 4 above)
Comment (MartinSchmidt) [reply chart/values.yaml:3]: remove the tag or use {name}:{tag} in the code.

File: chart/Chart.yaml [hunk 6]
```diff
@@ -0,0 +1,18 @@
+apiVersion: v2
+name: mukube-ingress
+description: A Helm chart for setting up Traefik as ingress controller for Kubernetes
+
+# A chart can be either an 'application' or a 'library' chart.
```
Comment (MartinSchmidt) [inline chart/Chart.yaml:5]: Remove all the boilerplate comments.

**Review by MartinSchmidt — CHANGES_REQUESTED** [review]
There are a series of comments I agree should be changed.

**Review by jacobcrawford — APPROVED** [review]
Looks good! :D

**Review by MartinSchmidt — APPROVED** [review]

**Pull request merged**